llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -1137,8 +1139,12 @@ IRBuilder<>::InsertPoint OpenMPIRBuilder::createParallel(
   // Change the location to the outer alloca insertion point to create and
   // initialize the allocas we pass into the parallel region.
   Builder.restoreIP(OuterAllocaIP);
-  AllocaInst *TIDAddr = Builder.CreateAlloca(Int32, nullptr, "tid.addr");
-  AllocaInst *ZeroAddr = Builder.CreateAlloca(Int32, nullptr, "zero.addr");
+  unsigned AllocaAddressSpace =
+      Config.isTargetDevice() ? 0 : M.getDataLayout().getAllocaAddrSpace();
```

Comment:
See above.

Reply:
Without this change the outlined function will have the following signature:
define internal void @__omp_offloading_fd00_2b03043__QQmain_l30..omp_par(**ptr addrspace(5)** noalias noundef %tid.addr, **ptr addrspace(5**) noalias noundef %zero.addr, **ptr addrspace(5**) %0)
(Tested on AMD GPU target)

Reply:
Fixed.